### DISCUSSION TITLE
Vitest can't find library with no types but I have a .d.ts file for it

### DISCUSSION THREAD

**Discussion by abcd-ca** [vitest-dev/vitest#2571] [Q&A] [answered]
I have a project and it compiles and runs. I am using the 'luxon-business-days' library which comes with no type definitions. I added a .d.ts file to declare a module and indicate the types for it which has helped my project compile and run but Vitest can't seem to find it.

Normally I can just do `import {DateTime} from 'luxon-business-days'` then ex. `DateTime.now()`. When I run my test file that calls a function in my project that uses the module, I get 

> Cannot read properties of undefined (reading 'now')

I did not create a vite.config.ts file in my root for vitest but I do have a src/client/vite.config.ts but it has no `test` section. 

Can someone help explain what is happening to cause the issue so that I might have a starting point for troubleshooting? It is the only library that vitest has an issue with – my other tests that don't depend on 'luxon-business-days' are running and passing without issue.

Thank you.

**Comment by sheremet-va** [comment, marked as answer]
Probably this: https://vitest.dev/config/#deps-interopdefault

**Comment by abcd-ca** [comment]
@sheremet-va That was it, thanks! 

For the benefit of others: In my case I have a project with client and server code and my `/src/client` has a `vite.config.ts` but that's really only for using vite for the front-end app. I wanted to unit test code from both `/src/client`, `/src/server` and `/src/shared` so I created `/vitest.config.ts` (as per https://vitest.dev/config/#configuration)

which contains,

```typescript
import { defineConfig, configDefaults } from 'vitest/config'

export default defineConfig({
  test: {
    deps: {
      ...configDefaults,
      interopDefault: true
    }
  },
})
```
